packages/tabs/src/TabTitle/TabTitle.styles.ts

```diff
@@ -1,16 +1,90 @@
-import { css } from '@leafygreen-ui/emotion';
+import { css, cx } from '@leafygreen-ui/emotion';
 import { Theme } from '@leafygreen-ui/lib';
 import { palette } from '@leafygreen-ui/palette';
 import {
   BaseFontSize,
+  borderRadius,
   fontFamilies,
   fontWeights,
+  spacing,
   transitionDuration,
   typeScales,
 } from '@leafygreen-ui/tokens';
 
-const TAB_TITLE_BODY_1_HEIGHT = 44;
-const TAB_TITLE_BODY_2_HEIGHT = 52;
+const BODY_1_HEIGHT = 44;
+const BODY_2_HEIGHT = 52;
+const MAX_WIDTH = 300;
+const SMALL_HEIGHT = 32;
+
+const baseStyles = css`
+  font-family: ${fontFamilies.default};
+  font-weight: ${fontWeights.medium};
+  position: relative;
+  display: inline-flex;
+  flex-direction: column;
+  align-items: flex-start;
+  justify-content: center;
+  max-width: ${MAX_WIDTH}px;
+  padding: ${spacing[300]}px ${spacing[400]}px;
+  background-color: transparent;
+  border: 0;
+  margin: 0;
+  text-decoration: none;
+  transition: ${transitionDuration.default}ms color ease-in-out;
+
+  &:focus:not(:disabled) {
+    outline: none;
+    font-weight: ${fontWeights.bold};
+  }
+
+  // We create a pseudo element that's the width of the bolded text
+  // This way there's no layout shift on hover when the text is bolded.
+  &:before {
+    content: attr(data-text);
+    height: 0;
+    font-weight: ${fontWeights.bold};
+    visibility: hidden;
+    overflow: hidden;
+    user-select: none;
+    pointer-events: none;
+  }
+
+  &:after {
+    content: '';
+    position: absolute;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    height: 4px;
+    border-radius: ${borderRadius[100]}px ${borderRadius[100]}px 0 0;
+    transition: all ${transitionDuration.default}ms ease-in-out;
+    background-color: transparent;
+    transform: scaleX(0.8);
+  }
+
+  &:hover:after {
+    transform: scaleX(0.95);
+  }
+
+  &:active:after {
+    &:after {
+      transform: scaleX(1);
+    }
+  }
+`;
+
+export const baseFontSizeStyles: Record<BaseFontSize, string> = {
```

Comment:
nit: These styles exist already in `Typography`, minus the height.

https://github.com/mongodb/leafygreen-ui/blob/94b4e7fa19473636f68d7e4689a44a4abadf51f7/packages/typography/src/styles.tsx#L16-L25